java/analyze-hotspots/src/main/java/com/esri/arcgisruntime/sample/analyzehotspots/AnalyzeHotspotsFragment.java

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright 2019 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.esri.arcgisruntime.sample.analyzehotspots;
+
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.design.widget.FloatingActionButton;
+import android.support.v4.app.Fragment;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Toast;
+import com.esri.arcgisruntime.concurrent.Job;
+import com.esri.arcgisruntime.concurrent.ListenableFuture;
+import com.esri.arcgisruntime.geometry.Point;
+import com.esri.arcgisruntime.geometry.SpatialReferences;
+import com.esri.arcgisruntime.layers.ArcGISMapImageLayer;
+import com.esri.arcgisruntime.layers.Layer;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.Viewpoint;
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.tasks.geoprocessing.*;
+
+import java.util.concurrent.ExecutionException;
+
+public class AnalyzeHotspotsFragment extends Fragment {
```

Comment:
I think we should still keep the actual geoprocessing logic in the MainActivity. At least in the context of the samples I think it makes sense to keep it in MainActivity. That way this code would also be immediately visible to the user in the SampleViewer. @TADraeseke would be able to give you some more details on SampleViewer requirements.

I think it's ok to factor out the dialogs into their own DialogFragment implementations.

Reply:
So we have two choices of retaining state between configuration changes:
1. Implement a Fragment and call `setRetainInstance(true)` on it.
2. Keep the logic in the MainActivity and specify `android:configChanges="orientation| screenSize |keyboardHidden"` for it in the manifest.

Since our samples usually don't have complicated UI that would have to be updated when we handle configuration changes ourselves, I would go for option 2 as this would keep the logic in the MainActivity, which means that code would also show up in the Sample Viewer. @TADraeseke - can you please comment on whether that's a requirement for the Sample Viewer - i.e. that the main code of the sample needs to be in MainActivity in order to show up in the Sample Viewer?

From my POV the Fragment would make sense if we had to show different UI components in the MainActivity but usually our samples aren't that complex.

Reply:
Sorry its taken me so long to get to this. Thanks both for this. This is a long running issue with samples...where to find the balance between a primary goal of readability of runtime code and a secondary goal of showing best practice with runtime API within Android's API.

I also like option 2, and we used to use this on most samples as a convenient get out of jail free card...but at some point we thought this was showing bad practice and removed it from all of our samples.

On one hand we have simple API calls for handling long running jobs (pause, cancel and progress listener etc) which we want to show being used. However, if they're not handled in a lifecycle aware way, they just run off into an eventual error (if the device rotates mid job for example).

I personally think we should use a cheat like configChanges in this case, and implement one sample (maybe even this one) as a demo of a life cycle best practice with long running parts of our APIs (jobs in particular).

Reply:
👍 for option 2 and one (or more) separate sample(s) that specifically discuss any aspects of life cycle events. These new samples could incorporate toolkit components that we might implement, for better support of restoring MapView/SceneView state after a destroy/re-create (MapFragment/MapViewModel).